fedimint-recurringd/src/db.rs

```diff
@@ -0,0 +1,185 @@
+use std::collections::HashMap;
+
+use fedimint_core::config::FederationId;
+use fedimint_core::core::OperationId;
+use fedimint_core::db::{AutocommitError, Database, IDatabaseTransactionOpsCoreTyped};
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::secp256k1::rand::thread_rng;
+use fedimint_core::secp256k1::PublicKey;
+use fedimint_core::{impl_db_lookup, impl_db_record};
+use fedimint_ln_client::recurring::{PaymentCodeId, PaymentCodeRootKey, RecurringPaymentProtocol};
+use futures::stream::StreamExt;
+use lightning_invoice::Bolt11Invoice;
+use rand::Rng;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash, Ord, PartialOrd, Encodable, Decodable)]
+pub struct FederationDbPrefix([u8; 16]);
+
+impl FederationDbPrefix {
+    pub fn random() -> FederationDbPrefix {
+        FederationDbPrefix(thread_rng().gen())
+    }
+
+    fn prepend(&self, byte: u8) -> Vec<u8> {
+        let mut full_prefix = Vec::with_capacity(17);
+        full_prefix.push(byte);
+        full_prefix.extend(&self.0);
+        full_prefix
+    }
+}
+
+async fn load_federation_clients(db: &Database) -> Vec<(FederationId, FederationDbPrefix)> {
+    let mut dbtx = db.begin_transaction_nc().await;
+    dbtx.find_by_prefix(&FederationClientPrefix)
+        .await
+        .map(|(k, v)| (k.federation_id, v.db_prefix))
+        .collect::<Vec<_>>()
+        .await
+}
+
+pub fn open_client_db(db: &Database, db_prefix: FederationDbPrefix) -> Database {
+    db.with_prefix(db_prefix.prepend(DbKeyPrefix::ClientDB as u8))
+}
+
+pub async fn try_add_federation_database(
+    db: &Database,
+    federation_id: FederationId,
+    db_prefix: FederationDbPrefix,
```

Comment:
I think you should probably just use the entire `FederationId` instead of the prefix. With the p